File: Apps/gm-web-app/src/app/stocks/stock-recommendation.ts
export class StockRecommendation {
    constructor(public date: Date,
        public symbol: string,
        public systemId: string,
        public sentiment: number,
        public prediction: number,
        public lowPrediction: number,
        public previousClose: number,
        public predictedPriceTarget: number,
        public predictedLowTarget: number) {

    }

    public getRgb(intensity: number) {
        const range = 255 - intensity;
        const b = intensity;
        let g = 0;
        let r = 0;
        if(this.sentiment > 50.0) {
            r = intensity;
            g = ((this.sentiment - 50.0) / 50.0)*range + intensity; 
        } else {
            g = intensity;
            r = ((50.0 - this.sentiment) / 50.0) * range + intensity; 
        }
        return 'rgb(' + r + ',' + g + ',' + b + ')';
    }

    public getHsl(intensity: number) {
        let h = 105;
        let s = this.sentiment;
        let l = intensity;
        if(this.sentiment < 50.0) {
            h = 0;
            s = (50 - this.sentiment) * 2.0;
        } else {
            s = (this.sentiment - 50.0) * 2.0;
        }
        return 'hsl(' + h + ',' + s + '%,' + l + '%)';
    }
}

File: Apps/gm-flutter-app/gimmillions/lib/models/stock-recommendation.dart
import 'package:flutter/cupertino.dart';
import 'package:gimmillions/models/stock-data.dart';

class StockRecommendation {
  DateTime date;
  String symbol;
  String systemId;
  double sentiment;
  double confidence;
  double prediction;
  double lowPrediction;
  double previousClose;
  double predictedPriceTarget;
  double predictedLowTarget;
  StockData stockData;

  StockRecommendation(this.date, this.symbol, this.systemId, this.sentiment, this.confidence, this.prediction,
      this.lowPrediction, this.previousClose, this.predictedPriceTarget, this.predictedLowTarget, this.stockData);

  String get concensus {
    if (sentiment > 90) {
      return 'STRONG BUY';
    } else if (sentiment >= 75) {
      return "BUY";
    } else if (sentiment > 25) {
      return "HOLD";
    } else if (sentiment > 10) {
      return "SELL";
    } else {
      return "STRONG SELL";
    }
  }

  Color getRgb(int intensity) {
    var range = 255 - intensity;
    var b = intensity.toDouble();
    var g = 0.0;
    var r = 0.0;
    if (this.sentiment > 50.0) {
      r = intensity.toDouble();
      g = ((this.sentiment - 50.0) / 50.0) * range + intensity;
    } else {
      g = intensity.toDouble();
      r = ((50.0 - this.sentiment) / 50.0) * range + intensity;
    }
    return Color.fromRGBO(r.toInt(), g.toInt(), b.toInt(), 1);
  }

  Color getRgbo(int intensity, double opacity) {
    var range = 255 - intensity;
    var b = intensity.toDouble();
    var g = 0.0;
    var r = 0.0;
    if (this.sentiment > 50.0) {
      r = intensity.toDouble();
      g = ((this.sentiment - 50.0) / 50.0) * range + intensity;
    } else {
      g = intensity.toDouble();
      r = ((50.0 - this.sentiment) / 50.0) * range + intensity;
    }
    return Color.fromRGBO(r.toInt(), g.toInt(), b.toInt(), opacity);
  }

  factory StockRecommendation.fromJson(Map<String, dynamic> json) {
    return StockRecommendation(
        DateTime.parse(json['date']),
        json['symbol'],
        json['systemId'],
        json['sentiment'].toDouble(),
        json['confidence'].toDouble(),
        json['prediction'].toDouble(),
        json['lowPrediction'].toDouble(),
        json['previousClose'].toDouble(),
        json['predictedPriceTarget'].toDouble(),
        json['predictedLowTarget'].toDouble(),
        StockData.fromJson(json['lastData']));
  }
}
